hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/JpaTarget.java

```diff
@@ -149,40 +142,50 @@ public class JpaTarget extends AbstractJpaNamedEntity implements Target, EventAw
     @JoinColumn(name = "assigned_distribution_set", nullable = true, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_assign_ds"))
     private JpaDistributionSet assignedDistributionSet;
 
+    // set default request controller attributes to true, because we want to request them the first time
+    @Column(name = "request_controller_attributes", nullable = false)
+    private boolean requestControllerAttributes = true;
+
+    @OneToOne(fetch = FetchType.LAZY, mappedBy = "target", orphanRemoval = true)
+    @PrimaryKeyJoinColumn
+    private JpaAutoConfirmationStatus autoConfirmationStatus;
+
+    @ManyToOne(fetch = FetchType.LAZY, targetEntity = JpaTargetType.class)
+    @JoinColumn(name = "target_type", foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_relation_target_type"))
+    private TargetType targetType;
+
+    @CascadeOnDelete
+    @ManyToMany(targetEntity = JpaTargetTag.class)
+    @JoinTable(
+            name = "sp_target_target_tag",
+            joinColumns = {
+                    @JoinColumn(name = "target", nullable = false, updatable = false, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_targ_targtag_target")) },
+            inverseJoinColumns = {
+                    @JoinColumn(name = "tag", nullable = false, updatable = false, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_targ_targtag_tag"))
+            })
+    private Set<TargetTag> tags;
+
     /**
-     * Read only on management API. Are committed by controller.
+     * Supplied / committed by the controller. Read-only via management API.
      */
     @CascadeOnDelete
     @ElementCollection
     @Column(name = "attribute_value", length = Target.CONTROLLER_ATTRIBUTE_VALUE_SIZE)
     @MapKeyColumn(name = "attribute_key", nullable = false, length = Target.CONTROLLER_ATTRIBUTE_KEY_SIZE)
-    @CollectionTable(name = "sp_target_attributes", joinColumns = {
-            @JoinColumn(name = "target_id", nullable = false, updatable = false) }, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_targ_attrib_target"))
+    @CollectionTable(
+            name = "sp_target_attributes",
+            joinColumns = {
+                    @JoinColumn(name = "target_id", nullable = false, updatable = false) }, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_targ_attrib_target"))
     private Map<String, String> controllerAttributes;
 
-    @ManyToOne(fetch = FetchType.LAZY, optional = true, targetEntity = JpaTargetType.class)
-    @JoinColumn(name = "target_type", nullable = true, updatable = true, foreignKey = @ForeignKey(value = ConstraintMode.CONSTRAINT, name = "fk_target_relation_target_type"))
```

Comment:
Are we sure we want to remove TargetType reference from a target ?

Reply:
not removed - just moved down to group type/tags/attrbutes and metadata on single place as "metadata" for the target